server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java

```diff
@@ -438,7 +438,7 @@ protected void failEngine(IOException cause) {
             handler.sendFiles(store, metas.toArray(new StoreFileMetaData[0]), () -> 0);
             fail("exception index");
         } catch (RuntimeException ex) {
-            assertNull(ExceptionsHelper.unwrapCorruption(ex));
+            assertNotNull(ExceptionsHelper.unwrapCorruption(ex));
```

Comment:
This now finds the underlying corruption exception that is supressed on the RuntimeException